### PR TITLE
added covariant PagedResult interface

### PR DESCRIPTION
**Pull request by tarurar**

### CODE REVIEW

File: src/Lykke.AzureStorage/Tables/Paging/PagedResult.cs [hunk 1]
```diff
@@ -3,20 +3,26 @@
...
+        PagingInfo PagingInfo { get; }
+    }
+
+    public class PagedResult<T>: IPagedResult<T>
+    {
+        private readonly IEnumerable<T> _result;
+
         public PagingInfo PagingInfo { get; }
 
         public PagedResult(IEnumerable<T> items = null, PagingInfo pagingInfo = null)
```
Comment (KonstantinRyazantsev) [inline src/Lykke.AzureStorage/Tables/Paging/PagedResult.cs]: ```items = null``` may lead to NullReferenceException in ```GetEnumerator()```, so it's bad default value.

File: src/Lykke.AzureStorage/Tables/Paging/PagedResult.cs [hunk 2]
```diff
@@ -3,20 +3,26 @@
...
 namespace Lykke.AzureStorage.Tables.Paging
 {
-    public class PagedResult<T>: IEnumerable<T>
+    public interface IPagedResult<out T> : IEnumerable<T>
     {
-        public IEnumerable<T> Result { get; }
+        PagingInfo PagingInfo { get; }
+    }
+
+    public class PagedResult<T>: IPagedResult<T>
```
Comment (KonstantinRyazantsev) [inline src/Lykke.AzureStorage/Tables/Paging/PagedResult.cs]: Is ```PagedResult``` class have to be public? Maybe better to remove it from the public API.

File: src/Lykke.AzureStorage/Tables/Paging/PagedResult.cs [hunk 3]
```diff
@@ -3,20 +3,26 @@
 
 namespace Lykke.AzureStorage.Tables.Paging
 {
-    public class PagedResult<T>: IEnumerable<T>
+    public interface IPagedResult<out T> : IEnumerable<T>
```
Comment (KonstantinRyazantsev) [inline src/Lykke.AzureStorage/Tables/Paging/PagedResult.cs]: Better to follow the rule: one type - one file

File: src/Lykke.AzureStorage/Tables/Paging/PagedResult.cs (hunk 1 above)
Comment (tarurar) [reply src/Lykke.AzureStorage/Tables/Paging/PagedResult.cs]: adde initialisation in case parameter is null

**Pull request merged**